### PR TITLE
Adapt FWCore/PythonUtilities to python3

### PR DESCRIPTION
**Pull request by fabiocos**
#### PR description:

Apply futurize to ```FWCore/PythonUtilities``` and adapt ```edmDumpEventContent``` to python3.

#### PR validation:

Tested locally both in the standard and python3 version of master.

### CODE REVIEW

**Comment by fabiocos** [comment]
please test

File: FWCore/PythonUtilities/python/LumiList.py [hunk 1]
```diff
@@ -73,12 +77,12 @@ def __init__(self, filename = None, lumis = None, runsAndLumis = None, runs = No
         if isinstance(runsAndLumis, list):
             queued = {}
             for runLumiList in runsAndLumis:
-                for run, lumis in runLumiList.items():
+                for run, lumis in list(runLumiList.items()):
```
Comment (makortel) [inline FWCore/PythonUtilities/python/LumiList.py]: Here and in many cases below: why is a `list` created here? The iteration should work fine in both py2 and py3 with the return value of `items()` (or `keys()`). (or am I missing something?)

File: FWCore/PythonUtilities/scripts/edmDumpEventContent [hunk 2]
```diff
@@ -42,8 +45,8 @@ def expandFilename (filename, options):
     stdout, stderr = getOutput ('edmFileUtil -d %s' % filename)
     if stdout:
         return stdout.strip()
-    raise RuntimeError, "Could not successfully run 'edmFileUtil.' Error %s" \
-          % stderr
+    raise_(RuntimeError, "Could not successfully run 'edmFileUtil.' Error %s" \
+          % stderr)
```
Comment (makortel) [inline FWCore/PythonUtilities/scripts/edmDumpEventContent]: I wonder why not
```python
    raise RuntimeError("Could not successfully run 'edmFileUtil.' Error %s" \
          % stderr)
```

File: FWCore/PythonUtilities/python/LumiList.py [hunk 3]
```diff
@@ -242,7 +246,7 @@ def getLumis(self):
         Return the list of pairs representation
         """
         theList = []
-        runs = self.compactList.keys()
+        runs = list(self.compactList.keys())
         runs = sorted(run, key=int)
```
Comment (makortel) [inline FWCore/PythonUtilities/python/LumiList.py]: (possibly not for this PR) I wonder if this should be
```python
runs = sorted(runs, key=int)
```
given that there is no local variable `run`.

**Comment by fabiocos** [comment]
@makortel thanks for your comments, futurize adds several non necessary updates besides needed ones. I have done a cleaning of the code, preserving functionalities. There was also a genuine bug by futurize in the modification of the check on urllib (breaking runTheMatrix...).

**Comment by fabiocos** [comment]
code-checks

**Comment by fabiocos** [comment]
@smuzaffar @Dr15Jones @makortel further comments or suggestions?

File: FWCore/PythonUtilities/scripts/edmDumpEventContent (hunk 2 above)
Comment (makortel) [reply FWCore/PythonUtilities/scripts/edmDumpEventContent]: Shouldn't the parentheses be `raise RuntimeError("...)`?

File: FWCore/PythonUtilities/scripts/edmDumpEventContent [hunk 4]
```diff
@@ -60,11 +63,11 @@ class Branch (object):
             self.type = match.group (1)
         else:
             # this isn't a type we're interested in
-            raise ValueError, "Not edm::Wrapper"
+            raise ValueError("Not edm::Wrapper")
         name = trailingDotRE.sub ('', branchInfo.GetName())
         pieces = underscoreRE.split (name)
         if len (pieces) != 4:
-            raise ValueError, "%s not formatted as expected" % name
+            raise (ValueError, "%s not formatted as expected" % name)
```
Comment (makortel) [inline FWCore/PythonUtilities/scripts/edmDumpEventContent]: Also here `raise ValueError("...)`

File: FWCore/PythonUtilities/scripts/edmDumpEventContent [hunk 5]
```diff
@@ -84,8 +87,8 @@ class Branch (object):
                 found = True
                 break
         if not found:
-            raise ValueError, "%s does not match regexs provided" \
-                  % self.fullname
+            raise (ValueError, "%s does not match regexs provided" \
+                  % self.fullname)
```
Comment (makortel) [inline FWCore/PythonUtilities/scripts/edmDumpEventContent]: And here `raise ValueError("...)`

File: FWCore/PythonUtilities/scripts/edmDumpEventContent [hunk 6]
```diff
@@ -218,12 +221,12 @@ if __name__ == "__main__":
...
     # root file.  When we've loaded the file, we're turn it back
     oldStderr = os.dup( sys.stderr.fileno() )
-    newStderr = file ( '/dev/null', 'w')
+    newStderr = open ( '/dev/null', 'w')
     os.dup2( newStderr.fileno(), sys.stderr.fileno() )
     rootfile = ROOT.TFile.Open( filename )
     os.dup2( oldStderr, sys.stderr.fileno() )
     if not rootfile:
-        raise RuntimeError, "Could not open file '%s'." % filename
+        raise (RuntimeError, "Could not open file '%s'." % filename)
```
Comment (makortel) [inline FWCore/PythonUtilities/scripts/edmDumpEventContent]: And here `raise RuntimeError("...)`

**Comment by fabiocos** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**